### PR TITLE
Nisp 1661 - Personal max scenarios

### PR DESCRIPTION
**Pull request by roundrobin99**
Preparing for early code review

### CODE REVIEW

File: app/uk/gov/hmrc/nisp/views/account.scala.html [hunk 1]
```diff
@@ -56,140 +56,254 @@
 
 @defining(Some(user)) { implicit userOption =>
     @nispMain(userLoggedIn = true, browserTitle = Messages("nisp.main.title"),
-                pageScripts=Some(pageScripts), articleClasses = Some("mainpage"), analyticsAdditionalJs = Some(analyticsAdditionalJs)) {
+              articleClasses = Some("mainpage"), analyticsAdditionalJs = Some(analyticsAdditionalJs)) {
+
+    <h1 class="heading-large">@Messages("nisp.main.h1.title")</h1>
+
+@spSummary.forecast.scenario
```
Comment (dspork) [inline app/uk/gov/hmrc/nisp/views/account.scala.html]: This should be removed once dev work is finished.

File: app/uk/gov/hmrc/nisp/views/account_forecastonly.scala.html [hunk 2]
```diff
@@ -30,15 +30,15 @@
...
+        'dimension8':  '@spSummary.forecast.forecastAmount.week.toInt',
         'dimension10': '@spSummary.numberOfQualifyingYears',
         'dimension11': '@spSummary.numberOfGaps',
         'dimension12': '@spSummary.numberOfGapsPayable',
         'dimension13': '@spSummary.yearsToContributeUntilPensionAge',
         'dimension14': '@spSummary.contractedOutFlag',
         'dimension15': '@spSummary.statePensionAge',
         'dimension16': '@spSummary.copeAmount.week',
-        'dimension21': '@spSummary.forecastOnlyFlag',
+        'dimension21': '@spSummary.forecast.scenario',
```
Comment (dspork) [inline app/uk/gov/hmrc/nisp/views/account_forecastonly.scala.html]: We agreed with Paul to replace the "contextMessage" with scenario. So remove it to dimension7, we should just get rid of dimension21

File: app/uk/gov/hmrc/nisp/views/scenarios/reachedMax.scala.html [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+* limitations under the License.
+*@
+
+@import uk.gov.hmrc.nisp.controllers.routes
+@import uk.gov.hmrc.play.views.formatting.Dates
+@import uk.gov.hmrc.nisp.models.SPSummaryModel
+
+@****************************************************@
+@*                                                  *@
+@*  Scenarios (Ahmed, Derek, Robert and Susan)      *@
```
Comment (dspork) [inline app/uk/gov/hmrc/nisp/views/scenarios/reachedMax.scala.html]: This documentation needs changed

**Comment by dspork** [comment]
Is the personal max info you added to the test data accurate? That can be another ticket if it's not.
There's merge conflicts that need resolved

File: app/uk/gov/hmrc/nisp/views/account_forecastonly.scala.html (hunk 2 above)
Comment (roundrobin99) [inline app/uk/gov/hmrc/nisp/views/account_forecastonly.scala.html]: Sounds good.

File: app/uk/gov/hmrc/nisp/views/scenarios/reachedMax.scala.html (hunk 3 above)
Comment (roundrobin99) [inline app/uk/gov/hmrc/nisp/views/scenarios/reachedMax.scala.html]: Yup

**Comment by roundrobin99** [comment]
@dspork No the test data are not accurate, do they have to ?

**Comment by dspork** [comment]
@roundrobin99 I'll happily add a ticket to the backlog about accurate test data, it's not negatively impacting anything it's just a nice to have.

File: app/uk/gov/hmrc/nisp/controllers/AccountController.scala [hunk 4]
```diff
@@ -58,6 +60,18 @@ trait AccountController extends NispFrontendController with AuthorisedForNisp wi
   val customAuditConnector: CustomAuditConnector
   val applicationConfig: ApplicationConfig
 
+  def showCope: Action[AnyContent] = AuthorisedByAny.async { implicit user => implicit request =>
+    isFromPertax.flatMap { isPertax =>
+      val nino = user.nino.getOrElse("")
+      val authenticationProvider = getAuthenticationProvider(user.authContext.user.confidenceLevel)
+      nispConnector.connectToGetSPResponse(nino).map {
+        case SPResponseModel(Some(spSummary: SPSummaryModel), None) =>
```
Comment (dspork) [inline app/uk/gov/hmrc/nisp/controllers/AccountController.scala]: This needs to check if they can see this with `getABTest` and `Redirect` back to the account page if they can't.

File: app/uk/gov/hmrc/nisp/utils/Breadcrumb.scala [hunk 5]
```diff
@@ -42,7 +42,8 @@ trait Breadcrumb {
       "voluntarycontribs" -> ((URLEncoder.encode(Messages("nisp.breadcrumb.nirecord.voluntaryContrib"), "UTF-8"), routes.NIRecordController.showVoluntaryContributions().url)),
       "gapsandhowtocheck" -> ((URLEncoder.encode(Messages("nisp.breadcrumb.nirecord.gapsandhowtocheck"), "UTF-8"), routes.NIRecordController.showGapsAndHowToCheckThem().url)),
       "exclusion" -> ((URLEncoder.encode(Messages("nisp.breadcrumb.excluded"), "UTF-8"), routes.ExclusionController.showSP().url)),
-      "exclusionni" -> ((URLEncoder.encode(Messages("nisp.breadcrumb.excluded"), "UTF-8"), routes.ExclusionController.showNI().url))
+      "exclusionni" -> ((URLEncoder.encode(Messages("nisp.breadcrumb.excluded"), "UTF-8"), routes.ExclusionController.showNI().url)),
+      "cope" -> ((URLEncoder.encode(Messages("nisp.breadcrumb.cope"), "UTF-8"), routes.AccountController.show().url))
```
Comment (dspork) [inline app/uk/gov/hmrc/nisp/utils/Breadcrumb.scala]: I know the customer will not see this URL but it should still have the cope one in to be correct.

**Comment by dspork** [comment]
Closing as work has begun on a new branch,

**Pull request closed without merging**